pkg/client/scale.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"github.com/openshift/origin/pkg/api/latest"
+)
+
+type scales struct {
+	oc *Client
+	kc     *kclient.Client
+	ns     string
+}
+
+func newScales(oc *Client, kc *kclient.Client, namespace string) *scales {
+	return &scales{
+		oc: oc,
+		kc: kc,
+		ns: namespace,
+	}
+}
+
+// Get takes the reference to scale subresource and returns the subresource or error, if one occurs.
+func (c *scales) Get(kind string, name string) (result *extensions.Scale, err error) {
```

Comment:
> He's implementing https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/scale.go#L29, which is used by https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/horizontal.go#L99

```go
type delegatingScaler struct{
    dcClient DeploymentConfigInterface
    kubeScaler kclient.Scaler
}

func (s *delegatingScaler) Get(kind string, name string) (*extensions.Scale, error){
    switch (kind){
        "DeploymentConfig":
            return s.dcClient.GetScale(name)
    }

    return s.kubeScaler.Get(kind, name)
}
```

That keeps the list explicit, ensures that the client libraries provide solid types, prevents URL building against resources that aren't ours (belong to kube), allows proper error messages (instead of 403's or 404's) for origin types, and with upstream changes would allow a better location/delegation model.